### PR TITLE
Fix for failing OSD distribution build

### PR DESCRIPTION
**Pull request by rishabh6788**
Signed-off-by: Rishabh Singh <sngri@amazon.com>

### Description
This PR fixes the failing distribution build for OSD and updates the manifest to use the new `dashboards-notifications` repository. 

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: scripts/default/opensearch-dashboards/build.sh [hunk 1]
```diff
@@ -72,6 +72,13 @@ if [ "$PLATFORM" = "windows" ]; then
     NVM_CMD="volta install node@`cat ../OpenSearch-Dashboards/.nvmrc` && volta install yarn@`jq -r '.engines.yarn' ../OpenSearch-Dashboards/package.json`"
 fi
 
+helper_cmd=$(jq -r '.scripts.plugin_helpers' package.json)
+if [ "$helper_cmd" != null ]; then
+        helper_string=plugin_helpers
+else
+        helper_string=plugin-helpers
```
Comment (peterzhuamazon) [inline scripts/default/opensearch-dashboards/build.sh:79]: Spacing can be fixed a bit to 4 spaces.

Comment (peterzhuamazon) [inline scripts/default/opensearch-dashboards/build.sh:79]: Could you use ALL_CAPITAL for both var name?
Thanks.

File: scripts/default/opensearch-dashboards/build.sh [hunk 2]
```diff
@@ -72,6 +72,13 @@ if [ "$PLATFORM" = "windows" ]; then
     NVM_CMD="volta install node@`cat ../OpenSearch-Dashboards/.nvmrc` && volta install yarn@`jq -r '.engines.yarn' ../OpenSearch-Dashboards/package.json`"
 fi
 
+HELPER_CMD=$(jq -r '.scripts.plugin_helpers' package.json)
+if [ "$HELPER_CMD" != null ]; then
+    HELPER_STRING=plugin_helpers
+else
+    HELPER_STRING=plugin-helpers
+fi
```
Comment (gaiksaya) [inline scripts/default/opensearch-dashboards/build.sh:80]: Is this a fast fix that all dashboards team needs to make a change in their package.json? Changing config during build can be dangerous.

Comment (peterzhuamazon) [reply scripts/default/opensearch-dashboards/build.sh:80]: There is no consistency on all the dashboards on how to call plugin helper.
Not sure which repo starts using plugin-helpers and some follows but some dont.
At this point this is a quick fix but the permanent fix is to let everyone use plugin_helpers.

**Review by peterzhuamazon — APPROVED** [review]

**Pull request merged**